### ISSUE TITLE
Unable to run

### ISSUE THREAD

**Issue by Bencor29** [jaredallard/ccdocker#7]
Hi, your project seems to be abandoned but I try my chance !

I got an error on running ccdocker (craftos-1.7's image pulled from fresh installed ccdockerd)
<details>
<summary>Error screenshot</summary>

![Error screenshot](https://i.imgur.com/crmSpWE.png)

</details>

I tried with non-minified ccdocker so I find the line:
<details>
<summary>Error screenshot</summary>

![Error screenshot](https://i.imgur.com/fQLm5lh.png)

</details>

File & Line :
https://github.com/jaredallard/ccdocker/blob/6268ef94187ee65722bbd395c3fd1f89e381c1cd/lib/docker.lua#L193

Hope you could help me

**Comment by jaredallard** [comment]
I'll take a look, but it's been _years_ so I don't really remember this code base

**Comment by Bencor29** [comment]
I have an idea !

https://github.com/jaredallard/ccdocker/blob/6268ef94187ee65722bbd395c3fd1f89e381c1cd/lib/docker.lua#L193

I think the error is thrown by `ipairs()` because he receive null. Why does he receive null ?
`this.fo` respond to the image parsed as JSON. 
Image: https://github.com/cswarm/ccdockerd/blob/master/images/craftos/1.7
The problem is, there is no keys called "c". But, in `docker.lua` there is no key called `children`.
My question is, did you, or someone else, modify the key name and didn't modify the image ?

Is it a possible answer ?

**Comment by Bencor29** [comment]
I tried to copy the 1.7 image and to rename all "children" keys to "c".
I got a different error:
<details>
<summary>Error</summary>

![Error](https://i.imgur.com/Zj0XxCr.png)
</details>

**Comment by Bencor29** [comment]
Ah ah! I found the problem !!
Stop looking for it, I'm writing the explanation and the PR

**Comment by Bencor29** [comment]
When I tried to rename "children" to "c", I was not totaly wrong.
The new error was not really easy to understand.
I search in the code to find when the error is thrown. And I found:
https://github.com/jaredallard/ccdocker/blob/6268ef94187ee65722bbd395c3fd1f89e381c1cd/lib/docker.lua#L799-L805

`o` is a string containing the script to be executed (`/rom/programs/shell` here).
The error says `string:60`, so in file `/rom/programs/shell` at char 60 (not line, it's a string) there is `local parentShell = shell`.

So... `shell` is null ?
ccdocker give a new env to the script with new values link multishell:
https://github.com/jaredallard/ccdocker/blob/6268ef94187ee65722bbd395c3fd1f89e381c1cd/lib/docker.lua#L562
But shell is not given. I added the line `["shell"] = deepcopy(shell),` after multishell and, magic ! It's working !!
